include/libtorrent/disk_io_thread.hpp

```diff
@@ -528,8 +558,13 @@ namespace libtorrent
 		// shutting down. This last thread is responsible for cleanup
 		std::atomic<int> m_num_running_threads;
 
-		// the actual threads running disk jobs
-		std::vector<std::thread> m_threads;
+		// std::mutex to protect the m_generic_io_jobs and m_hash_io_jobs lists
+		mutable std::mutex m_job_mutex;
+
+		job_queue m_generic_io_jobs;
```

Comment:
maybe it would be good with a comment here, explaining that different kinds of jobs are posted to different pools. and maybe which jobs go to which pool, and maybe even why.